core/capabilities/ccip/launcher/deployment.go

```diff
@@ -41,41 +41,57 @@ func (c pluginRegistry) TransitionFrom(prevPlugins pluginRegistry) error {
 		return fmt.Errorf("current pluginRegistry or prevPlugins have more than 4 instances: len(prevPlugins): %d, len(currPlugins): %d", len(prevPlugins), len(c))
 	}
 
+	prevOracles := mapset.NewSet[ocrtypes.ConfigDigest](maps.Keys(prevPlugins)...)
+	currOracles := mapset.NewSet[ocrtypes.ConfigDigest](maps.Keys(c)...)
+
+	var ops = make([]syncAction, 0, 8)
+	for digest := range prevOracles.Difference(currOracles).Iterator().C {
+		ops = append(ops, syncAction{
+			command: closeAction,
+			oracle:  prevPlugins[digest],
+		})
+	}
+
+	for digest := range currOracles.Difference(prevOracles).Iterator().C {
+		ops = append(ops, syncAction{
+			command: openAction,
+			oracle:  c[digest],
+		})
+	}
+
 	var wg sync.WaitGroup
 	var mu sync.Mutex
-	// This shuts down instances that were present previously, but are no longer needed
-	for digest, oracle := range prevPlugins {
-		if _, ok := c[digest]; !ok {
-			wg.Add(1)
-			go func(o cctypes.CCIPOracle) {
-				defer wg.Done()
-				if err := o.Close(); err != nil {
+	for _, op := range ops {
+		wg.Add(1)
+		go func(op syncAction) {
```

Comment:
We should maybe declare `op:= op` to avoid unexpected behavior because the goroutine may use the final value of the loop variable

Reply:
talked offline, but this should be fixed: 
https://go.dev/blog/loopvar-preview
> For Go 1.22, we plan to change for loops to make these variables have per-iteration scope instead of per-loop scope